hazelcast/src/main/java/com/hazelcast/internal/partition/ChunkSupplier.java

```diff
@@ -34,7 +34,7 @@ public interface ChunkSupplier extends Iterator<Operation> {
     /**
      * @param isEndOfChunk boolean supplier to signal end of chunk.
      */
-    default void signalEndOfChunkWith(BooleanSupplier isEndOfChunk) {
+    default void signalEndOfChunkWith(Predicate isEndOfChunk) {
```

Comment:
Is it possible to use a typed `Predicate` here and in other places?

Reply:
done.